parsers/s01-parse/LePresidente/grafana-logs.yaml

```diff
@@ -1,16 +1,28 @@
 onsuccess: next_stage
-#debug: false
+debug: true
 name: LePresidente/grafana-logs
 filter: "evt.Parsed.program == 'grafana'"
 description: "Parse grafana logs"
 nodes:
   - grok:
-      pattern: 't=%{TIMESTAMP_ISO8601:timestamp} level=%{GREEDYDATA:loglevel} msg="Invalid username or password" error="%{GREEDYDATA:error}" remote_addr=%{IP:remote_ip}'
+      pattern: 't=%{TIMESTAMP_ISO8601:timestamp} level=%{GREEDYDATA:loglevel} msg=("Invalid username or password"|Unauthorized) error="%{GREEDYDATA:error}" remote_addr=%{IP:remote_ip}'
       apply_on: message
       statics:
         - meta: log_type
           value: grafana_failed_auth
-
+  - filter: UnmarshalJSON(evt.Parsed.message, evt.Unmarshaled, "grafana") in ["", nil]
+    nodes:
+    - filter: evt.Unmarshaled.grafana.msg == 'Invalid username or password'
+      nodes:
+        - grok:
```

Comment:
Do you want to use GROK to make sure they are compliant with the pattern?

Because if that is not the case you can just

```
      statics:
        - meta: log_type
          value: grafana_failed_auth
        - parsed: remote_ip
          expression: evt.Unmarshaled.grafana.remote_addr
        - parsed: timestamp
          expression: evt.Unmarshaled.grafana.t
```